src/resources/rmd/patch.R

```diff
@@ -268,3 +268,12 @@ valid_path = function(prefix, label) {
 }
 assignInNamespace("valid_path", valid_path, ns = "knitr")
 
+
+# add special language comment options support in knitr
+if (knitr_has_yaml_chunk_options()) {
+  knitr_comment_chars <- knitr:::comment_chars
+  knitr_comment_chars$ojs <- "//"
+  knitr_comment_chars$mermaid <- "%%"
```

Comment:
FYI I just added them to **knitr**: https://github.com/yihui/knitr/commit/3dcfac4780accc3791855b298a556297f62dc73d

Reply:
Answered fast. Just realizing why I really did the patched in Quarto. 

Based on what we see in https://github.com/quarto-dev/quarto-cli/pull/7804 is that doing this in **knitr** means that `yaml.code` option now contains the options and not just `code`

I fixed that part in the PR here:
https://github.com/quarto-dev/quarto-cli/blob/f9ed45429ca157e055210c9c1ff7df10e4850f2e/src/resources/rmd/execute.R#L55-L62

but this means new knitr 1.46 will have impact (potential breakage) with Quarto 1.3. 

That is the probem with making change in **knitr** and potential impact on Quarto version 🤷‍♂️

Reply:
Okay, it's clear to me now. Thanks for the explanation!

Reply:
https://github.com/yihui/knitr/commit/b037a31888b101af00e4988a1444cadc31456587 should avoid the breakage. It also means this #7799 fix would be unnecessary, unless the next Quarto release is scheduled to be earlier than the next **knitr** release. The next version of **knitr** will be released relatively soon (perhaps in the next week or two).

Reply:
> It also means this https://github.com/quarto-dev/quarto-cli/pull/7799 fix would be unnecessary,

I believe the fix would make Quarto 1.4 works with older **knitr**, and your fix in **knitr** insure that newer knitr also work correctly. 

If I revert #7799, it would mean that a **knitr** update is required by the user, but this could be considered too. 

@cscheid thoughts ?